lib/msal-core/src/utils/ErrorUtils.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthCache } from "../cache/AuthCache";
+import { AuthError } from "../error/AuthError";
+/**
+ * @hidden
+ */
+export class ErrorUtils {
+
+    static errorHandler(storage: AuthCache, errorFunc: AuthError): AuthError {
+        storage.resetTempCacheItems();
+        return errorFunc;
```

Comment:
An alternative design would be to throw the error, instead of returning it.

Reply:
I actually did that first. There is some code in UAA that needs a return path (`openPopupWindow` related) and throwing an error here is causing compiler issues in that function. There could be a different solution, lets talk tomorrow.